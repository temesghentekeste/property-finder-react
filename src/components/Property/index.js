import React from 'react';

const Property = () => (
  <div>
    <h1>Property</h1>
  </div>
);

export default Property;
