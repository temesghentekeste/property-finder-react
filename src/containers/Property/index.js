/* eslint-disable no-unused-vars */
import { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { WaveLoading } from 'react-loadingg/lib';
import { getPropertyAsync } from '../../redux/propertySlice';
import PropertyComponent from '../../components/PropertyComponent';

const Property = () => {
  const { id } = useParams();
  const dispatch = useDispatch();
  const { loading, error, property } = useSelector((state) => state.property);

  useEffect(() => {
    dispatch(getPropertyAsync(id));
  }, [dispatch]);

  if (loading || loading === null || property === null) {
    return <WaveLoading />;
  }

  if (error) {
    return <h1>{error}</h1>;
  }
  let name;
  let description;
  let featuredImage;
  let address;
  let isForRent;
  let price;
  let attributes;
  if (
    typeof property === 'object'
    && property !== null
    && property.data.attributes
  ) {
    name = property.data.attributes.name;
    address = property.data.attributes.address;
    price = property.data.attributes.monthly_price;
    description = property.data.attributes.description;
    featuredImage = property.data.attributes.featured_image;
    isForRent = property.data.attributes.is_for_rent;
  }
  return (
    property && (
      <PropertyComponent
        name={name}
        address={address}
        description={description}
        image={featuredImage}
        isForRent={isForRent}
        price={price}
      />
    )
  );
};

export default Property;
