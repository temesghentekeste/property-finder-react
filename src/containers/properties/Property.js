import React from 'react'

const Property = () => {
    return (
        <div>
            <h1>Property</h1>
        </div>
    )
}

export default Property
