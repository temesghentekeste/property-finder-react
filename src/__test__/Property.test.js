import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Property from '../components/PropertyComponent';

let getByTestId;

beforeEach(() => {
  const component = render(
    <Router>
      <Property
        name="Test Property"
        address="Test Address"
        price="500"
        description="Lorem ipsum"
        image="https://ichef.bbci.co.uk/news/976/cpsprodpb/6127/production/_117317842_065760657.jpg"
        isForRent={true}
      />
    </Router>
  );
  getByTestId = component.getByTestId;
});

it('renders the property component', () => {
  const component = renderer
    .create(
      <Router>
        <Property
        name="Test Property"
        address="Test Address"
        price="500"
        description="Lorem ipsum"
        image="https://ichef.bbci.co.uk/news/976/cpsprodpb/6127/production/_117317842_065760657.jpg"
        isForRent={true}
      />
      </Router>
    )
    .toJSON();

  expect(component).toMatchSnapshot();
});
