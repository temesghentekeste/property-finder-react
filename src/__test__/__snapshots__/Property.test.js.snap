// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the property component 1`] = `
<section
  className="propertyContainer"
>
  <div
    className="makeStyles-root-1"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-2 mui-fixed MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
        >
          Properties Finder |
           
          Test Property
        </h6>
      </div>
    </header>
    <div
      className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-6"
    >
      <div
        className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-7 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <div
          className="makeStyles-drawerHeader-8"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </span>
          </button>
        </div>
        <hr
          className="MuiDivider-root"
        />
        <ul
          className="MuiList-root MuiList-padding"
        >
          <a
            className="makeStyles-drawerH1-9"
            href="/properties"
            onClick={[Function]}
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemIcon-root"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  style={
                    Object {
                      "color": "#4caf50",
                    }
                  }
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9.93 13.5h4.14L12 7.98zM20 2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-4.05 16.5l-1.14-3H9.17l-1.12 3H5.96l5.11-13h1.86l5.11 13h-2.09z"
                  />
                </svg>
              </div>
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  All Properties
                </span>
              </div>
            </div>
            <hr
              className="MuiDivider-root"
            />
          </a>
          <a
            className="makeStyles-drawerH1-9"
            href="/favorites"
            onClick={[Function]}
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemIcon-root"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </div>
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Your Favorites
                </span>
              </div>
            </div>
            <hr
              className="MuiDivider-root"
            />
          </a>
          <a
            className="makeStyles-drawerH1-9"
            href="/dashboard"
            onClick={[Function]}
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemIcon-root"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 14h4v-4H4v4zm0 5h4v-4H4v4zM4 9h4V5H4v4zm5 5h12v-4H9v4zm0 5h12v-4H9v4zM9 5v4h12V5H9z"
                  />
                </svg>
              </div>
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Dashboard
                </span>
              </div>
            </div>
            <hr
              className="MuiDivider-root"
            />
          </a>
          <a
            className="makeStyles-drawerH1-9"
            href="/"
            onClick={[Function]}
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemIcon-root"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                  />
                </svg>
              </div>
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Sign Out
                </span>
              </div>
            </div>
            <hr
              className="MuiDivider-root"
            />
          </a>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="propertyCard"
    data-testid="property"
  >
    <img
      alt="Test Property"
      src="https://ichef.bbci.co.uk/news/976/cpsprodpb/6127/production/_117317842_065760657.jpg"
    />
    <div
      className="propertyCard__info"
    >
      <p
        data-testid="property-name"
      >
        <span>
          Name: 
        </span>
        Test Property
      </p>
      <p>
        <span>
          Address: 
        </span>
        Test Address
      </p>
      <p>
        <span>
          Price: 
        </span>
        $
        500
      </p>
      <p>
        <span>
          Description: 
        </span>
        Lorem ipsum
      </p>
      <p>
        <span>
          Availability: 
        </span>
        Available
      </p>
    </div>
  </div>
</section>
`;
